libs/model/src/chain-events/ChainEventCreated.command.ts

```diff
@@ -0,0 +1,135 @@
+import {
+  EvmCommunityStakingEventSignatures,
+  EvmNamespaceFactoryEventSignatures,
+  EvmRecurringContestEventSignatures,
+  EvmSingleContestEventSignatures,
+  logger as loggerFactory,
+  type Command,
+} from '@hicommonwealth/core';
+import { config, equalEvmAddresses } from '@hicommonwealth/model';
+import * as schemas from '@hicommonwealth/schemas';
+import { commonProtocol as cp } from '@hicommonwealth/shared';
+import { Hmac, createHmac } from 'crypto';
+import { fileURLToPath } from 'url';
+
+const __filename = fileURLToPath(import.meta.url);
+const logger = loggerFactory(__filename);
+
+// TODO: how do we handle chain re-orgs
+
+function anyAddressEqual(
+  addresses: string[],
+  addressToCompare: string,
+): boolean {
+  for (const address of addresses) {
+    if (equalEvmAddresses(address, addressToCompare)) return true;
+  }
+  return false;
+}
+
+export function verifyAlchemySignature(req: any) {
```

Comment:
probably better in a middleware file

Reply:
Generally, I would agree but this is very specific to the Alchemy Webhook auth flow which right now only consists of a single route. I would contend that it should stay here for ease of access/discoverability since it won't be reused anywhere else for the time being. This whole Alchemy flow is somewhat convoluted so I want to limit the spread of changes as much as possible.